kratos/solving_strategies/schemes/residual_based_bdf_scheme.h

```diff
@@ -0,0 +1,591 @@
+        if (D.size1() != 0) { // if D matrix declared
+            noalias(LHS_Contribution) += D * mBDF[0];
+        }
+    }
+
+    /**
+     * @brief It adds the dynamic RHS contribution of the objects 
+     * \f[ \mathbf{b} - \mathbf{M} a - \mathbf{D} v \f]
+     * @param rObject The object to compute
+     * @param RHS_Contribution The dynamic contribution for the RHS
+     * @param D The damping matrix
+     * @param M The mass matrix
+     * @param rCurrentProcessInfo The current process info instance
+     */
+        
+    template <typename TObjectType>
+    void TemplateAddDynamicsToRHS(
+        TObjectType rObject,
+        LocalSystemVectorType& RHS_Contribution,
+        LocalSystemMatrixType& D,
+        LocalSystemMatrixType& M,
+        ProcessInfo& rCurrentProcessInfo
+        )
+    {
+        const std::size_t this_thread = OpenMPUtils::ThisThread();
+        
+        // Adding inertia contribution
+        if (M.size1() != 0) {
+            rObject->GetSecondDerivativesVector(mVector.dot2un0[this_thread], 0);
+            noalias(RHS_Contribution) -= prod(M, mVector.dot2un0[this_thread]);
+        }
+
+        // Adding damping contribution
+        if (D.size1() != 0) {
+            rObject->GetFirstDerivativesVector(mVector.dotun0[this_thread], 0);
+            noalias(RHS_Contribution) -= prod(D, mVector.dotun0[this_thread]);
+        }
+    }
+    
+    /**
+     * @brief It adds the dynamic RHS contribution of the elements 
+     * \f[ \mathbf{b} - \mathbf{M} a - \mathbf{D} v \f]
+     * @param pElement The element to compute
+     * @param RHS_Contribution The dynamic contribution for the RHS
+     * @param D The damping matrix
+     * @param M The mass matrix
+     * @param rCurrentProcessInfo The current process info instance
+     */
+
+    void AddDynamicsToRHS(
+        Element::Pointer pElement,
+        LocalSystemVectorType& RHS_Contribution,
+        LocalSystemMatrixType& D,
+        LocalSystemMatrixType& M,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        TemplateAddDynamicsToRHS<Element::Pointer>(pElement, RHS_Contribution, D, M, rCurrentProcessInfo);
+    }
+
+    /**
+     * @brief It adds the dynamic RHS contribution of the condition
+     *  \f[ RHS = f_{ext} - \ddot{u}_{n0} \mathbf{M} + \dot{u}_{n0} \mathbf{D} + u_{n0} \mathbf{K} \f]
+     * @param pCondition The condition to compute
+     * @param RHS_Contribution The dynamic contribution for the RHS
+     * @param D The damping matrix
+     * @param M The mass matrix
+     * @param rCurrentProcessInfo The current process info instance
+     */
+
+    void AddDynamicsToRHS(
+        Condition::Pointer pCondition,
+        LocalSystemVectorType& RHS_Contribution,
+        LocalSystemMatrixType& D,
+        LocalSystemMatrixType& M,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        TemplateAddDynamicsToRHS<Condition::Pointer>(pCondition, RHS_Contribution, D, M, rCurrentProcessInfo);
```

Comment:
`TemplateAddDynamicsToRHS(pCondition, RHS_Contribution, D, M, rCurrentProcessInfo);` does not work?

Reply:
I like to specify the template argument, yes, in this case can be automatically detected, but...I  have my things

Reply:
ok